src/Mutator/NodeMutationGenerator.php

```diff
@@ -129,7 +130,11 @@ private function generateForMutator(Node $node, IgnoreMutator $mutator, array $m
             return $mutations;
         }
 
-        foreach ($mutator->mutate($node) as $mutationByMutatorIndex => $mutatedNode) {
+        // It is important to not rely on the keys here. It might otherwise result in some elements
+        // being overridden, see https://3v4l.org/JLN73
+        $mutatedNodes = iterator_to_array($mutator->mutate($node), false);
+
+        foreach ($mutatedNodes as $mutationByMutatorIndex => $mutatedNode) {
```

Comment:
So it is correct that you propose we basically ignore all keys from mutators? I don't think this is a major problem, but it can be non-obvious.

Reply:
Alternatively we can increment `$mutationByMutatorIndex` as we go.

Reply:
Yes, I find it a bit tricky in this case as it may really be non-obvious, but truth be told it's something you always need to be wary of with _any_ generator...